pkg/cli/login/loginoptions.go

```diff
@@ -497,6 +561,38 @@ func (o *LoginOptions) SaveConfig() (bool, error) {
 	return created, nil
 }
 
+func (o *LoginOptions) extractIssuerURLForOIDC() (string, error) {
```

Comment:
where's the code that already does this today? Can we reuse it?

Reply:
It is in this function https://github.com/openshift/library-go/blob/8c7cc0e6362857307c2df1e53c86fd830e3b871e/pkg/oauth/tokenrequest/request_token.go#L156-L205 . Since we have nothing to do with `OsinConfig` and `oauthMetadataEndpoint` should be exported, there needs to be some refactorings in library-go. But we have to backport this PR to 4.15 and it is not feasible to make changes on library-go and cherry-pick it to 4.15 within this short time frame. 

I can only promise that I'll refactor library-go and remove this function from `oc` in 4.17.